modules/yieldbotBidAdapter.js

```diff
@@ -0,0 +1,614 @@
+  },
+
+  get pageviewDepth() {
+    const cookieName = '__ybotpvd';
+    let cookieValue = parseInt(this.getCookie(cookieName), 10) || 0;
+    cookieValue++;
+    this.setCookie(cookieName, cookieValue, this._sessionTimeout, '/');
+    return cookieValue;
+  },
+
+  get lastPageviewId() {
+    const cookieName = '__ybotlpvi';
+    let cookieValue = this.getCookie(cookieName);
+    return cookieValue || '';
+  },
+
+  set lastPageviewId(id) {
+    const cookieName = '__ybotlpvi';
+    return this.setCookie(cookieName, id, this._sessionTimeout, '/');
+  },
+
+  get lastPageviewTime() {
+    const cookieName = '__ybotlpv';
+    let cookieValue = this.getCookie(cookieName);
+    return cookieValue ? parseInt(cookieValue, 10) : 0;
+  },
+
+  set lastPageviewTime(ts) {
+    const cookieName = '__ybotlpv';
+    return this.setCookie(cookieName, ts, this._sessionTimeout, '/');
+  },
+
+  /**
+   * Get/set the request base url used to form bid, ad markup and impression requests.
+   * @param {string} [prefix] the bidder request base url
+   * @returns {string} the request base Url string
+   * @memberof module:YieldbotBidAdapter
+   */
+  urlPrefix: function(prefix) {
+    const cookieName = '__ybotc';
+    const pIdx = prefix ? prefix.indexOf(':') : -1;
+    const url = pIdx !== -1 ? document.location.protocol + prefix.substr(pIdx + 1) : null;
+    let cookieValue = url || this.getCookie(cookieName);
+    if (!cookieValue) {
+      cookieValue = '//i.yldbt.com/m/';
+    }
+    this.setCookie(cookieName, cookieValue, this._sessionTimeout, '/');
+    return cookieValue;
+  },
+
+  /**
+   * Bidder identifier code.
+   * @type {string}
+   * @constant
+   * @memberof module:YieldbotBidAdapter
+   */
+  get code() { return 'yieldbot'; },
+
+  /**
+   * A list of aliases which should also resolve to this bidder.
+   * @type {string[]}
+   * @constant
+   * @memberof module:YieldbotBidAdapter
+   */
+  get aliases() { return []; },
+
+  /**
+   * @property {MediaType[]} [supportedMediaTypes]: A list of Media Types which the adapter supports.
+   * @constant
+   * @memberof module:YieldbotBidAdapter
+   */
+  get supportedMediaTypes() { return ['banner']; },
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @returns {boolean} True if this is a valid bid, and false otherwise.
+   * @memberof module:YieldbotBidAdapter
+   */
+  isBidRequestValid: function(bid) {
+    let invalidSizeArray = false;
+    if (utils.isArray(bid.sizes)) {
+      const arr = bid.sizes.reduce((acc, cur) => acc.concat(cur), []).filter((item) => {
+        return !utils.isNumber(item);
+      });
+      invalidSizeArray = arr.length !== 0;
+    }
+    const ret = bid &&
+            bid.params &&
+            utils.isStr(bid.params.psn) &&
+            utils.isStr(bid.params.slot) &&
+            !invalidSizeArray &&
+            !!bid.params.psn &&
+      !!bid.params.slot;
```

Comment:
why are psn and slot checked twice?

Reply:
Over zealous perhaps.

It's not uncommon for there to be misconfiguration of parameters: `psn` and/or `slot` is seen to be an empty string.